### ISSUE TITLE
Implementation of editor

### ISSUE THREAD

**Issue by StephanStanisic** [StephanStanisic/simple-blog#4]
This is a open discussion on how to implement the editing functions of this plugin.

Currently this plugin uses a contenteditable area to receive user input. This is way different than the core implementation, so the summernote-editor plugin can't find this editable area.

My question now: Is there a way to make the editable areas of this plugin automatically include in the summernote plugin?

**Comment by robiso** [comment]
I think this could be a bigger issue, will look into it with the help of @slavenstancic.

**Comment by robiso** [comment]
@StephanStanisic, is there any way to change the content editable to textarea?

Summernote **does** support content editable, but it takes the HTML editing rights away from the user. That's why WCMS currently uses textarea, as the user has the freedom to enter HTML code in any way he wishes to.

Let me know if you need any help with the textarea (as we currently use jQuery to send the data from the textarea to the database.js).

**Comment by StephanStanisic** [comment]
Changing to a `textarea` would be no problem. 

What are you suggesting to let summernote take the textarea's?

**Comment by StephanStanisic** [comment]
Hmm, just realized that if we add `.editable` to the textarea summernote will just pick them up like they are from WonderCMS itself. Not sure what that will mean for the saving script of WonderCMS.

**Comment by robiso** [comment]
@StephanStanisic, sorry forgot to mention it. Not sure what's going to happen.
Let me know when you push an update, I'll run it for a test. :)

**Comment by robiso** [comment]
@StephanStanisic had any chance to play around with this? I'm in suspense by what could of happened with the saving part/db through Summernote.

**Comment by StephanStanisic** [comment]
@robiso I've tested + added summernote support from commit f89c76023925864c264b6d2246be219a45015741. 

It looks (client side) if it can find any `script` tags containing `summernote` (so it matches both summernote-editor and summernote-air-editor).  If that's the case it adds a listener to the onBlur event of summernote and links that with the internal save mechanism of the blog.

One thing I found is that with the plugins is that their path is relative to the current document, and only works for paths in the root directory. Since the blogposts aren't in the root path, the paths 404.

I've fixed this by passing the paths to `$Wcms->url()`, will prepare some PR's.
